gateway/reverse_proxy_test.go

```diff
@@ -408,6 +408,134 @@ func TestCircuitBreaker5xxs(t *testing.T) {
 	})
 }
 
+func TestCircuitBreakerEvents(t *testing.T) {
+	// Use this channel to capture webhook events:
+	triggeredEvent := make(chan apidef.TykEvent)
+
+	// Establish a simple HTTP server that takes webhook input and passes the event to above channel:
+	webHookServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		rawBody, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// Unmarshal webhook input, we're only interested in event's name:
+		var eventData map[string]interface{}
+		if err := json.Unmarshal(rawBody, &eventData); err != nil {
+			t.Fatal(err)
+		}
+		eventName, ok := eventData["event"].(string)
+		if !ok {
+			t.Fatal("invalid webhook input")
+		}
+		triggeredEvent <- apidef.TykEvent(eventName)
+	}))
+
+	// Establish another HTTP server to trigger CB behavior
+	// Uses a counter to send an error response on the 1st sample:
+	var sampleCount int
+	upstreamServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if sampleCount == 1 {
+			w.WriteHeader(500)
+			sampleCount++
+			return
+		}
+		w.WriteHeader(200)
+		sampleCount++
+	}))
+
+	ts := StartTest(nil)
+	defer ts.Close()
+
+	// Events to capture on this API, we use default webhook template:
+	events := map[apidef.TykEvent][]apidef.EventHandlerTriggerConfig{
+		EventBreakerTripped: {
+			{
+				Handler: EH_WebHook,
+				HandlerMeta: map[string]interface{}{
+					"method":        http.MethodPost,
+					"target_path":   webHookServer.URL,
+					"template_path": "templates/default_webhook.json",
+					"event_timeout": 10,
+				},
+			},
+		},
+		EventBreakerReset: {
+			{
+				Handler: EH_WebHook,
+				HandlerMeta: map[string]interface{}{
+					"method":        http.MethodPost,
+					"target_path":   webHookServer.URL,
+					"template_path": "templates/default_webhook.json",
+					"event_timeout": 10,
+				},
+			},
+		},
+	}
+
+	// Setup an API definition with CB settings and attach above event handlers:
+	ts.Gw.BuildAndLoadAPI(func(spec *APISpec) {
+		spec.Proxy.TargetURL = upstreamServer.URL
+		spec.Proxy.ListenPath = "/circuitbreaker/"
+		spec.CircuitBreakerEnabled = true
+		UpdateAPIVersion(spec, "v1", func(v *apidef.VersionInfo) {
+			err := json.Unmarshal([]byte(`[
+					{
+						"path": "test",
+						"method": "GET",
+						"threshold_percent": 0.1,
+						"samples": 1,
+						"return_to_service_after": 500
```

Comment:
I missed this one because `return_to_service_after` is actually in seconds, sub-second values in the `time.Sleep` that we use for the final event don't work, the minimum is 1 second.

Reply:
Minimum value is 1 second, docs ref: https://tyk.io/docs/planning-for-production/ensure-high-availability/circuit-breakers/#configure-with-the-api-definition